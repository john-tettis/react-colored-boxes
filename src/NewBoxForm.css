.new-box-form{
    box-sizing: border-box;
    padding:2rem;
    display:flex;
    height:100%;
    flex-direction:column;
    justify-content: space-around;
    background-color:antiquewhite;
}

.new-box-form-container{
    
    margin-bottom:2rem;
    
}
.new-box-form-container1{
    width:80%;
}
.new-box-form-container2{
    width:20%;
}

.new-box-form-container2 > *{
    margin-bottom:1rem;
}
