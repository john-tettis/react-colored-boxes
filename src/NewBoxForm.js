import React, {useState} from 'react'

import Slider from '@material-ui/core/Slider';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import './NewBoxForm.css'



const NewBoxForm=({createBox})=>{
    let [form,setForm] =useState({height:100, width:100,color:''})

    let handleSubmit=(e)=>{
        e.preventDefault()
        createBox(form)
    }
    let handleChange=(value,isHeight)=>{
        let dimension = isHeight ? 'height' : 'width';
        setForm((old)=>({...old, [dimension]:value}))
        //setForm(curr=>({...curr,[e.target.parentElement.id]:Number(e.target.firstChild.firstChild.innerText)}))
    }
    let handleColorChange = (e)=>{
        setForm(old=>({...old, color:e.target.value}))
    }


    return(
        <form onSubmit={handleSubmit}className='new-box-form'>
           <div className='new-box-form-container1'> 
               <div className='new-box-form-container'>
                <Slider
                        aria-labelledby="discrete-slider-always"
                        step={1}
                        valueLabelDisplay="on"
                        onChange={(e,value)=>handleChange(value,true)}
                        id='height'
                        type='range'
                        min={40}
                        max={200}
                        name='height'
                        value={form.height}
                />
                <Typography id="discrete-slider-always" gutterBottom>
                    Height
                </Typography>
                </div>
                <div className='new-box-form-container'>
                    <Slider
                            aria-labelledby="discrete-slider-always"
                            step={1}
                            valueLabelDisplay="on"
                            onChange={(ev,value)=>handleChange(value,false)}
                            id='width'
                            type='range'
                            min={40}
                            max={200}
                            name='width'
                            value={form.width}
                    />
                    <Typography id="discrete-slider-always" gutterBottom>
                        Width
                    </Typography>
                </div>
                
            </div>
            <div className='new-box-form-container2'>
                <Typography id="discrete-slider-always" gutterBottom>
                        Color
                </Typography>
                <input onChange={handleColorChange} id='color'type='color'name='color' value={form.color}></input>
            </div>
            
            <Button variant="outlined" onClick={handleSubmit}color="primary"  aria-label="create square">Add Square</Button>

        </form>
    )
}

export default NewBoxForm