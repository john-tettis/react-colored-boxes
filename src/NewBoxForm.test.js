import { render, screen } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test('renders NewBoxForm', () => {
  render(<NewBoxForm />);
});

test('renders NewBoxForm', () => {
    const {asFragment} = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot()
  });
