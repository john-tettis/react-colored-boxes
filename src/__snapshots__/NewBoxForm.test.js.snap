// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NewBoxForm 1`] = `
<DocumentFragment>
  <form
    class="new-box-form"
  >
    <div
      class="new-box-form-container1"
    >
      <div
        class="new-box-form-container"
      >
        <span
          class="MuiSlider-root MuiSlider-colorPrimary"
          id="height"
          type="range"
        >
          <span
            class="MuiSlider-rail"
          />
          <span
            class="MuiSlider-track"
            style="left: 0%; width: 37.5%;"
          />
          <input
            name="height"
            type="hidden"
            value="100"
          />
          <span
            aria-labelledby="discrete-slider-always"
            aria-orientation="horizontal"
            aria-valuemax="200"
            aria-valuemin="40"
            aria-valuenow="100"
            class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-open-7 PrivateValueLabel-thumb-6"
            data-index="0"
            role="slider"
            style="left: 37.5%;"
            tabindex="0"
          >
            <span
              class="PrivateValueLabel-offset-8 MuiSlider-valueLabel"
            >
              <span
                class="PrivateValueLabel-circle-9"
              >
                <span
                  class="PrivateValueLabel-label-10"
                >
                  100
                </span>
              </span>
            </span>
          </span>
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          id="discrete-slider-always"
        >
          Height
        </p>
      </div>
      <div
        class="new-box-form-container"
      >
        <span
          class="MuiSlider-root MuiSlider-colorPrimary"
          id="width"
          type="range"
        >
          <span
            class="MuiSlider-rail"
          />
          <span
            class="MuiSlider-track"
            style="left: 0%; width: 37.5%;"
          />
          <input
            name="width"
            type="hidden"
            value="100"
          />
          <span
            aria-labelledby="discrete-slider-always"
            aria-orientation="horizontal"
            aria-valuemax="200"
            aria-valuemin="40"
            aria-valuenow="100"
            class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-open-7 PrivateValueLabel-thumb-6"
            data-index="0"
            role="slider"
            style="left: 37.5%;"
            tabindex="0"
          >
            <span
              class="PrivateValueLabel-offset-8 MuiSlider-valueLabel"
            >
              <span
                class="PrivateValueLabel-circle-9"
              >
                <span
                  class="PrivateValueLabel-label-10"
                >
                  100
                </span>
              </span>
            </span>
          </span>
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          id="discrete-slider-always"
        >
          Width
        </p>
      </div>
    </div>
    <div
      class="new-box-form-container2"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        id="discrete-slider-always"
      >
        Color
      </p>
      <input
        id="color"
        name="color"
        type="color"
        value=""
      />
    </div>
    <button
      aria-label="create square"
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Add Square
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
